File: lib-util/src/main/java/code/util/NetworkUtil.java
package code.util;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkUtil {
    public static final String TAG = NetworkUtil.class.getSimpleName();

    public static boolean isAvailable(Context context) {
        ConnectivityManager cm =
                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        //ObfuscationStub3.inject();
        return activeNetwork != null
                && activeNetwork.isConnectedOrConnecting();
    }

    public static boolean isConnected(Context context) {
        ConnectivityManager cm =
                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
        //ObfuscationStub4.inject();
        return activeNetwork != null
                && activeNetwork.isConnected();
    }

}

File: lib-util/src/main/java/code/util/NetworkUtil.kt
package code.util

import android.annotation.SuppressLint
import android.content.Context
import android.net.ConnectivityManager

object NetworkUtil {
   private val TAG = NetworkUtil::class.java.simpleName

    @SuppressLint("MissingPermission")
    fun isAvailable(context: Context): Boolean {
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetwork = cm.activeNetworkInfo
        //ObfuscationStub3.inject();
        return (activeNetwork != null
                && activeNetwork.isConnectedOrConnecting)
    }

    @SuppressLint("MissingPermission")
    
    fun isConnected(context: Context): Boolean {
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val activeNetwork = cm.activeNetworkInfo
        //ObfuscationStub4.inject();
        return (activeNetwork != null
                && activeNetwork.isConnected)
    }
}